File: VB/MyLayoutControl/MyLayoutControl.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraLayout.Registrator
Imports DevExpress.XtraLayout
Imports DevExpress.LookAndFeel
Imports DevExpress.XtraLayout.Painting
Imports DevExpress.XtraLayout.ViewInfo
Imports DevExpress.Utils.Drawing
Imports DevExpress.Skins

Namespace WindowsApplication1
	<System.ComponentModel.DesignerCategory("")> _
	Public Class MyLayoutControl
		Inherits LayoutControl
		Public Event CustomDrawBackground As CustomDrawEventHandler
		Public Event CustomDrawTextArea As CustomDrawEventHandler
		Public Event CustomDrawControlsArea As CustomDrawEventHandler
		Public Event CustomDrawSelection As CustomDrawEventHandler

		Protected Overrides Function CreateILayoutControlImplementorCore() As LayoutControlImplementor
			Return New MyLayoutControlImplementor(Me)
		End Function


		#Region "OnCustomDrawBackground"
		''' <summary>
		''' Triggers the CustomDrawBackground event.
		''' </summary>
		Public Overridable Sub OnCustomDrawBackground(ByVal ea As CustomDrawEventArgs)
			RaiseEvent CustomDrawBackground(Me, ea)
		End Sub
		#End Region
		#Region "OnCustomDrawTextArea"
		''' <summary>
		''' Triggers the CustomDrawTextArea event.
		''' </summary>
		Public Overridable Sub OnCustomDrawTextArea(ByVal ea As CustomDrawEventArgs)
			RaiseEvent CustomDrawTextArea(Me, ea)
		End Sub
		#End Region
		#Region "OnCustomDrawControlsArea"
		''' <summary>
		''' Triggers the CustomDrawControlsArea event.
		''' </summary>
		Public Overridable Sub OnCustomDrawControlsArea(ByVal ea As CustomDrawEventArgs)
			RaiseEvent CustomDrawControlsArea(Me, ea)
		End Sub
		#End Region
		#Region "OnCustomDrawSelection"
		''' <summary>
		''' Triggers the CustomDrawSelection event.
		''' </summary>
		Public Overridable Sub OnCustomDrawSelection(ByVal ea As CustomDrawEventArgs)
			RaiseEvent CustomDrawSelection(Me, ea)
		End Sub
		#End Region
	End Class
End Namespace

File: CS/C#/MyLayoutControl/MyLayoutControl.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraLayout.Registrator;
using DevExpress.XtraLayout;
using DevExpress.LookAndFeel;
using DevExpress.XtraLayout.Painting;
using DevExpress.XtraLayout.ViewInfo;
using DevExpress.Utils.Drawing;
using DevExpress.Skins;

namespace WindowsApplication1
{
    [System.ComponentModel.DesignerCategory("")]
    public class MyLayoutControl : LayoutControl
    {
        public event CustomDrawEventHandler CustomDrawBackground;
        public event CustomDrawEventHandler CustomDrawTextArea;
        public event CustomDrawEventHandler CustomDrawControlsArea;
        public event CustomDrawEventHandler CustomDrawSelection;

        protected override LayoutControlImplementor CreateILayoutControlImplementorCore()
        {
            return new MyLayoutControlImplementor(this);
        }

    
        #region OnCustomDrawBackground
        /// <summary>
        /// Triggers the CustomDrawBackground event.
        /// </summary>
        public virtual void OnCustomDrawBackground(CustomDrawEventArgs ea)
        {
            if (CustomDrawBackground != null)
                CustomDrawBackground(this, ea);
        }
        #endregion
        #region OnCustomDrawTextArea
        /// <summary>
        /// Triggers the CustomDrawTextArea event.
        /// </summary>
        public virtual void OnCustomDrawTextArea(CustomDrawEventArgs ea)
        {
            if (CustomDrawTextArea != null)
                CustomDrawTextArea(this, ea);
        }
        #endregion
        #region OnCustomDrawControlsArea
        /// <summary>
        /// Triggers the CustomDrawControlsArea event.
        /// </summary>
        public virtual void OnCustomDrawControlsArea(CustomDrawEventArgs ea)
        {
            if (CustomDrawControlsArea != null)
                CustomDrawControlsArea(this, ea);
        }
        #endregion
        #region OnCustomDrawSelection
        /// <summary>
        /// Triggers the CustomDrawSelection event.
        /// </summary>
        public virtual void OnCustomDrawSelection(CustomDrawEventArgs ea)
        {
            if (CustomDrawSelection != null)
                CustomDrawSelection(this, ea);
        }
        #endregion
    }
}